conans/client/tools/settings.py

```diff
@@ -75,3 +75,16 @@ def valid_min_cppstd(conanfile, cppstd, gnu_extensions=False):
     except ConanInvalidConfiguration:
         return False
     return True
+
+
+def stdcpp_library(settings):
```

Comment:
For uniformity, I'd prefer to pass the `conanfile` here, even if we are not using anything but the `settings`.

Or, if it justifies it a little bit, we can output a warning if the settings object doesn't contain `compiler` 😄

Reply:
maybe we should check then if passed object is **settings** or **conanfile**, just like other tools are doing?

Reply:
No, they are not doing that check. They were first introduced with `settings` and then, when we realize we need to pass the `conanfile`, we implement that check so those tools were backward compatible. But now they are documented only with the `conanfile` variant.

This is one of the reasons why I vote to use always `conanfile`, eventually those tools need something else and we need to do that awful workaround to allow the double signature.

Reply:
okay, let's just check to accept conanfile instance then

Reply:
changed signature to accept conanfile